File: 프로그래머스/2/42584. 주식가격/주식가격.java
import java.util.*;

class Solution {
    public int[] solution(int[] prices) {
        
        Deque<int[]> que = new ArrayDeque<>();
        int len = prices.length;
        int answer[] = new int[len];
        
        for(int i=0; i<len; i++){
            while(!que.isEmpty() && que.peekLast()[0] > prices[i]){
                int out[] = que.pollLast();
                answer[out[1]] = i - out[1]; 
            }
            que.add(new int[] {prices[i], i});
        }
        
        while(!que.isEmpty()){
            int out[] = que.pollLast();
            answer[out[1]] = len-1-out[1];
        }
        
        return answer;
    }
}

File: 프로그래머스/2/42584. 주식가격/주식가격.js
function solution(prices) {
    
    const stack = [];
    const answer = [];
    const len = prices.length;
    
    for(let i=0; i<len; i++){
        while(stack.length && stack[stack.length-1][0] > prices[i]){
            var out = stack.pop();
            answer[out[1]] = i - out[1];
        }
        stack.push([prices[i], i]);
    }
    
    while(stack.length){
        out = stack.pop();
        answer[out[1]] = len-1-out[1];
    }
    
    return answer;
}